src/drivers/intel/ssp/ssp.c

```diff
@@ -741,20 +741,57 @@ static int ssp_get_hw_params(struct dai *dai,
 	return 0;
 }
 
-/* SSP hw params */
+/*
+ * enabled clock MCLK/BCLK to support codec that need MCLK/BCLK
+ * This is called before triiger(COMP_TRIGGER_START)
+ */
 static int ssp_hw_params(struct dai *dai,
 			 struct sof_ipc_stream_params  *params)
 {
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
+
 	dai_info(dai, "ssp_hw_params()");
 
+	/* Do nothing if clk_early_start is not set */
+	if (!ssp->params.clk_early_start)
+		return 0;
+
+	spin_lock(&dai->lock);
+
+	/* request mclk/bclk */
+	ssp_pre_start(dai);
```

Comment:
We probably need a change to the code in https://github.com/thesofproject/sof/blob/c7b75f536e9fcbaba6d647bfe02413a572be82c0/src/drivers/intel/ssp/ssp.c#L632, the component does not become active here so if we call ssp_pre_start a second time the protection does not work